docs/reference.md

```diff
@@ -2122,14 +2123,75 @@ protected:
 See "[Writing custom tracers](CookBook.md/#custom_tracer)" in the
 [Cook Book](CookBook.md) for an example.
 
-### <A name="typed_matcher"/>`trompeloeil::typed_matcher<T>`
+### <A name="typed_matcher"/> `trompeloeil::typed_matcher<T>`
 
 Convenience class available when writing custom matchers for a specific
 type. It inherits from [`trompeloeil::matcher`](#matcher_type).
 
 See "[Writing custom matchers](CookBook.md/#custom_matchers)" in the
 [Cook Book](CookBook.md) for examples.
 
+### <A named="movable_mock"/> `trompeloeil_movable_mock`
+
+By adding a public member type `trompeloeil_movable_mock` to your mock
+struct/class, you make it move constructible. Note that when a mock object is
+moved, any current expectations will be taken over by the newly constructed
+mock object, but the implicitly created lambdas associated with
+[**`.WITH()`**](reference.md/#WITH),
+[**`.SIDE_EFFECT()`**](reference.md/#SIDE_EFFECT),
+[**`.RETURN()`**](reference.md/#RETURN) and
+[**`.THROW()`**](reference.md/#THROW) and their `**LR_**` counter parts, will
+not themselves be moved. This means that if they refer to member variables
+in the mock objects, they will refer to dead store after the move. Also, keep
+in mind the lifetime of expectations. If you move a mock object that has
+active expectations, they should almost certainly be
+[**`NAMED_REQUIRE_CALL()`**](reference.md/#NAMED_REQUIRE_CALL),
+[**`NAMED_ALLOW_CALL()`**](reference.md/#NAMED_ALLOW_CALL) or
+[**`NAMED_FORBID_CALL()`**](reference.md/#NAMED_FORBID_CALL).
```

Comment:
What do you mean by "the ... lambdas ... will not themselves be moved"? Aren't the lambdas part of the expectations?

What problem can occur when using `REQUIRE_CALL` instead of `NAMED_REQUIRE_CALL`? In the non-named case I guess that the expectation is owned by the mock object so won't it be transferred to the new mock object on move-construction?

Reply:
I will need to think about how to phrase this better. This is actually quite wrong... Thanks.